File: Shantanu_Mamgain/Two-Pointer-Leetcode/Move-Zeroes.java
class Solution {
    public void swap(int[] nums, int i, int zero){
        int temp = nums[i];
        nums[i] = nums[zero];
        nums[zero] = temp;
    }
    public void moveZeroes(int[] nums) {
        int zero = -1;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0){
                zero = i;
                break;
            }
        }
      
        if(zero == -1){
            return;
        }

        for(int i = zero+1; i < nums.length; i++){
            if(nums[i] != 0){
                swap(nums, i, zero);
                zero ++;
            }
        }
    }
}

File: Lakshay Garg/Leetcode/Two_Pointers_Leetcode/Move_Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& v) {
        int n=v.size(),i=0,j=1;
        while(i<n && j<n){
            if(v[i]==0 && v[j]!=0){
                swap(v[i],v[j]);
                i++;
            }
            j++;
            if(v[i]!=0) i++;
        }
    }
};
